### PR TITLE
Add address state to company model

### PR DESCRIPTION
**Pull request by faissalMT**
### Description of change

Continuing from https://github.com/uktrade/data-hub-api/pull/3342 this adds the appropriate migrations and data model changes needed.  

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/company/models/company.py [hunk 1]
```diff
@@ -167,6 +167,14 @@ class GreatProfileStatus(models.TextChoices):
     address_1 = models.CharField(max_length=MAX_LENGTH, blank=True)
     address_2 = models.CharField(max_length=MAX_LENGTH, blank=True)
     address_town = models.CharField(max_length=MAX_LENGTH, blank=True)
+    address_state = models.CharField(max_length=MAX_LENGTH, blank=True)
+    address_state = models.ForeignKey(
```
Comment (tomkingston) [inline datahub/company/models/company.py:171]: These two fields have the same name -presumably you only want the second one?

File: datahub/company/models/company.py [hunk 2]
```diff
@@ -180,6 +188,15 @@ class GreatProfileStatus(models.TextChoices):
     registered_address_1 = models.CharField(max_length=MAX_LENGTH, blank=True)
     registered_address_2 = models.CharField(max_length=MAX_LENGTH, blank=True)
     registered_address_town = models.CharField(max_length=MAX_LENGTH, blank=True)
+    registered_address_state = models.CharField(max_length=MAX_LENGTH, blank=True)
+    registered_address_state = models.ForeignKey(
```
Comment (tomkingston) [inline datahub/company/models/company.py:192]: These two fields have the same name, presumably you only want the second one?

File: datahub/company/migrations/0108_create_company_state_base_model.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            model_name='company',
+            name='address_state',
+            field=models.CharField(blank=True, max_length=255),
+        ),
+        migrations.AddField(
+            model_name='company',
+            name='registered_address_state',
+            field=models.CharField(blank=True, max_length=255),
+        ),
+    ]
```
Comment (tomkingston) [inline datahub/company/migrations/0108_create_company_state_base_model.py:23]: I cannot see the need for this migration if you are just altering this field again in the next migration?

**Review by tomkingston — CHANGES_REQUESTED** [review]

File: datahub/company/models/company.py [hunk 4]
```diff
@@ -167,7 +167,13 @@ class GreatProfileStatus(models.TextChoices):
     address_1 = models.CharField(max_length=MAX_LENGTH, blank=True)
     address_2 = models.CharField(max_length=MAX_LENGTH, blank=True)
     address_town = models.CharField(max_length=MAX_LENGTH, blank=True)
-    address_county = models.CharField(max_length=MAX_LENGTH, blank=True)
```
Comment (ambrozic) [inline datahub/company/models/company.py:170]: do we want to remove `address_county` field?

File: datahub/company/models/company.py [hunk 5]
```diff
@@ -167,7 +167,13 @@ class GreatProfileStatus(models.TextChoices):
...
+    address_state = models.ForeignKey(
```
Comment (elcct) [inline datahub/company/models/company.py:170]: I am not sure if `address_state` is a correct name - is this field going to contain administrative areas other than states? Maybe `address_administrative_area` could better describe it, the same with `registered_address_`.

Comment (faissalMT) [reply datahub/company/models/company.py]: We've changed it to `address_area_code`, is that ok?

File: datahub/company/models/company.py [hunk 6]
```diff
@@ -180,6 +187,14 @@ class GreatProfileStatus(models.TextChoices):
     registered_address_1 = models.CharField(max_length=MAX_LENGTH, blank=True)
     registered_address_2 = models.CharField(max_length=MAX_LENGTH, blank=True)
     registered_address_town = models.CharField(max_length=MAX_LENGTH, blank=True)
+    registered_address_area_code = models.ForeignKey(
+        metadata_models.AdministrativeArea,
+        related_name='companies_with_registered_state',
```
Comment (ambrozic) [inline datahub/company/models/company.py:192]: [lets rename this to `companies_with_registered_address_area_code` for the sake of consistency albeit it being long and ugly.](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds)

File: datahub/company/models/company.py (hunk 5 above)
Comment (ambrozic) [reply datahub/company/models/company.py]: should we go with just `address_area` and `registered_address_area` ?

Comment (elcct) [reply datahub/company/models/company.py]: I agree that `address_area` is better.

**Review by ambrozic — APPROVED** [review]

**Review by elcct — APPROVED** [review]

**Pull request merged**